contracts/Core/StakeManager.sol

```diff
@@ -186,13 +186,15 @@ contract StakeManager is Initializable, StakeStorage, StateManager, Pause, Stake
         IStakedToken sToken = IStakedToken(staker.tokenAddress);
         require(sToken.balanceOf(msg.sender) >= sAmount, "Invalid Amount");
         uint256 totalSupply = sToken.totalSupply();
+        uint256 rAmount = _convertSRZRToRZR(sAmount, staker.stake, totalSupply);
+        uint256 prevStake = staker.stake;
 
-        if ((staker.stake - _convertSRZRToRZR(sAmount, staker.stake, totalSupply)) < minStake) {
+        if ((staker.stake - rAmount) < minStake) {
             // slither-disable-next-line reentrancy-events,reentrancy-no-eth
             rewardManager.giveInactivityPenalties(epoch, stakerId);
         }
 
-        uint256 rAmount = _convertSRZRToRZR(sAmount, staker.stake, totalSupply);
+        if(prevStake != staker.stake) rAmount = _convertSRZRToRZR(sAmount, staker.stake, totalSupply);
```

Comment:
hey cant this line moved after L195 only in if block ?
also we don't have to track prevStake imo
we can just make follwoing at L195, right ?

``rAmount = _convertSRZRToRZR(sAmount, staker.stake, totalSupply);``

with current impl, we will have redundant prevStake and if condition - L197 checked everytime

Reply:
oh ya correct. forgot about this